dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/TCPUtils.java

```diff
@@ -0,0 +1,276 @@
+import io.openliberty.netty.internal.ServerBootstrapExtended;
+import io.openliberty.netty.internal.exception.NettyException;
+import io.openliberty.netty.internal.impl.NettyFrameworkImpl;
+import io.openliberty.netty.internal.impl.NettyConstants;
+
+
+public class TCPUtils {
+
+    private static final TraceComponent tc = Tr.register(TCPUtils.class, TCPMessageConstants.NETTY_TRACE_NAME,
+            TCPMessageConstants.TCP_BUNDLE);
+    private static final int timeBetweenRetriesMsec = 1000; // make this non-configurable
+
+    /**
+     * Create a {@link ServerBootstrapExtended} for inbound TCP channels
+     * 
+     * @param framework
+     * @param tcpOptions
+     * @return
+     * @throws NettyException
+     */
+    public static ServerBootstrapExtended createTCPBootstrap(NettyFrameworkImpl framework,
+            Map<String, Object> tcpOptions) throws NettyException {
+        BootstrapConfiguration config = new TCPConfigurationImpl(tcpOptions, true);
+        ServerBootstrapExtended bs = new ServerBootstrapExtended();
+        bs.group(framework.getParentGroup(), framework.getChildGroup());
+        bs.channel(NioServerSocketChannel.class);
+        // apply the existing user config to the Netty TCP channel
+        bs.applyConfiguration(config);
+        ChannelInitializerWrapper tcpInitializer = new TCPChannelInitializerImpl(config);
+        bs.setBaseInitializer(tcpInitializer);
+        return bs;
+    }
+
+    /**
+     * Create a {@link BootstrapExtended} for outbound TCP channels
+     * 
+     * @param framework
+     * @param tcpOptions
+     * @return
+     * @throws NettyException
+     */
+    public static BootstrapExtended createTCPBootstrapOutbound(NettyFrameworkImpl framework,
+            Map<String, Object> tcpOptions) throws NettyException {
+        BootstrapConfiguration config = new TCPConfigurationImpl(tcpOptions, false);
+        BootstrapExtended bs = new BootstrapExtended();
+        bs.group(framework.getParentGroup());
+        bs.channel(NioSocketChannel.class);
+        // apply the existing user config to the Netty TCP channel
+        bs.applyConfiguration(config);
+        ChannelInitializerWrapper tcpInitializer = new TCPChannelInitializerImpl(config);
+        bs.setBaseInitializer(tcpInitializer);
+        return bs;
+    }
+
+    private static ChannelFuture open(NettyFrameworkImpl framework, AbstractBootstrap bootstrap,
+            final TCPConfigurationImpl config, String inetHost, int inetPort, ChannelFutureListener openListener,
+            final int retryCount) {
+
+        ChannelFuture oFuture = null;
+        if (inetHost.equals("*")) {
+            inetHost = NettyConstants.INADDR_ANY;
+        }
+        if (config.isInbound()) {
+            oFuture = ((ServerBootstrapExtended) bootstrap).bind(inetHost, inetPort);
+        } else {
+            oFuture = ((BootstrapExtended) bootstrap).connect(inetHost, inetPort);
+        }
+        final ChannelFuture openFuture = oFuture;
```

Comment:
why oFuture and openFuture too?

Reply:
OK

Reply:
Following up on this [here](https://github.com/OpenLiberty/open-liberty/issues/24840)